shotover-proxy/src/message/mod.rs

```diff
@@ -447,27 +451,200 @@ impl Value {
+                    Value::Set(typed_set)
+                }
+                ColType::Udt => {
+                    if let Some(ColTypeOptionValue::UdtType(ref list_type_option)) =
+                        spec.col_type.value
+                    {
+                        let len = list_type_option.descriptions.len();
+                        let decoded_udt = decode_udt(actual_bytes, len).unwrap();
+
+                        let udt = Udt::new(decoded_udt, list_type_option)
+                            .as_cassandra_type()
+                            .unwrap()
+                            .unwrap();
+
+                        let typed_udt = Value::create_udt(udt);
+
+                        Value::Udt(typed_udt)
+                    } else {
+                        panic!("Not a UDT. This indicates a bug in cassandra_protocol")
+                    }
+                }
+                ColType::Tuple => {
+                    if let Some(ColTypeOptionValue::TupleType(ref list_type_option)) =
+                        spec.col_type.value
+                    {
+                        let len = list_type_option.types.len();
+                        let decoded_tuple = decode_tuple(actual_bytes, len).unwrap();
+                        let tuple = Tuple::new(decoded_tuple, list_type_option)
+                            .as_cassandra_type()
+                            .unwrap()
+                            .unwrap();
+
+                        let typed_tuple = Value::create_tuple(tuple);
+                        Value::Tuple(typed_tuple)
+                    } else {
+                        panic!("Not a Tuple. This indicates a bug in cassandra_protocol")
+                    }
+                }
+                ColType::Custom => unimplemented!(),
+                ColType::Null => Value::NULL,
             }
         } else {
             Value::NULL
         }
     }
 
+    fn create_udt(collection: CassandraType) -> BTreeMap<String, Value> {
+        if let CassandraType::Udt(udt) = collection {
+            let mut values = BTreeMap::new();
+            udt.into_iter().for_each(|(key, element)| {
+                values.insert(key, Value::create_element(element));
+            });
+
+            values
+        } else {
+            panic!("Not a UDT. Only CassandraType::Udt should be passed to this method.");
+        }
+    }
+
+    #[allow(clippy::mutable_key_type)]
+    fn create_map(collection: CassandraType) -> BTreeMap<Value, Value> {
+        if let CassandraType::Map(map) = collection {
+            let mut value_list = BTreeMap::new();
+            for (key, value) in map.into_iter() {
+                value_list.insert(Value::create_element(key), Value::create_element(value));
+            }
+
+            value_list
+        } else {
+            panic!("Not a Map. Only CassandraType::Map should be passed to this method");
+        }
+    }
+
+    fn create_element(element: CassandraType) -> Value {
```

Comment:
Its unfortunate this duplicates a bunch of logic with build_value_from_cstar_col_type.
Is it not possible to turn the `spec: &ColSpec, data: &CBytes` into a `CassandraType` and call this instead?